### DISCUSSION TITLE
Como saber que podemos avançar nos estudos?

### DISCUSSION THREAD

**Discussion by ebraimcarvalho** [frontendbr/forum#1763] [Discussão]
Bom, às vezes sinto que preciso dominar bem (pelo menos 90%) do que estou estudando para poder avançar nos estudos. Exemplo: Eu quero aprender e desenvolver com React, mas não sei o quanto de JS Vanilla preciso saber pra poder avançar.
Para vocês, fazem mais sentido um estudo focado, um assunto de cada vez ou estudar assuntos relacionados paralelamente, em dias alternados, ex: seg: vanilla js, ter: react, qua: estrutura de dados e algoritmos, qui: vanilla js, sex: react...?

Pergunto isso porque vi um dia desses um conceito de Deep Work e Prática Deliberada que me fez pensar até que ponto focar em um assunto por vez é benéfico ou estou perdendo tempo...

**Comment by gabriel-hahn** [comment]
Para mim, focar em uma coisa de cada vez é mais importante para tu absorver o que tu esta estudando e depois disso colocar este conhecimento em prática. Por exemplo, vale mais a pena fazer um curso completo de React e depois construir um projeto do zero com uma ideia própria tua ou copiando algum projeto existente do que fazer mais um ou dois cursos de React sobre assuntos 'avançados' sem ao menos ter construído algo básico antes. Depois dessa 'base', tu parte para tópicos mais avançados do assunto ou parte para estudar uma outra coisa.
Referente a qualidade do código, por exemplo o teu JS enquanto constrói uma aplicação, acredito que isso vai melhorando com o tempo e tu vai encontrar problemas reais que só serão resolvidos com uma otimização do teu código JS, com uma estrutura diferente ou com alguma função dele que tu não conhece ainda (ai tu realmente memoriza aquilo e aprende como utilizar na prática).

**Comment by leandroruel** [comment]
o único jeito de avançar nos estudos é botando a mão na massa, tentar criar uma coisa avançada e quebrando a cabeça, se você vê que não sabe fazer determinada coisa, então sabe que tem que estudar aquilo, estudando e aprendendo, você vai resolver o problema e superar um obstáculo.
